### PR TITLE
connect: use exp backoff when waiting on consul envoy bootstrap

### PR DESCRIPTION
**Pull request by shoenig**
This PR wraps the use of the consul envoy bootstrap command in
an expoenential backoff closure, configured to timeout after 60
seconds. This is an increase over the current behavior of making
3 attempts over 6 seconds.

Should help with #10451

### CODE REVIEW

File: helper/exptime/backoff.go
Comment (shoenig) [inline helper/exptime/backoff.go:1]: We implement "exponential back-off" in a few places, but none of them generic and feature-full. It'd be nice to converge on a single implementation.

**Comment by shoenig** [comment]
I'm wondering if we should backport this

File: helper/exptime/backoff.go
Comment (tgross) [inline helper/exptime/backoff.go:63]: It looks like this alias is the only diff between `oss.indeed.com/go/libtime/decay` and this source file? While I understand and agree with not wanting to pull in the whole darn `libtime` library for this, the "derived work" being created here is an awfully minimal change. So for licensing safety, I think we're going to need to include the upstream [license file](https://github.com/indeedeng/libtime/blob/master/LICENSE) in this package.

File: helper/exptime/backoff.go
Comment (tgross) [inline helper/exptime/backoff.go:132]: This is fine because we should avoid creating more of a diff here, but "Dear `libtime` library authors, 2x per iteration is geometric backoff, not exponential." 😀

**Review by tgross — CHANGES_REQUESTED** [review]
LGTM once we get the license in there. The library seems like the right one to use; they're accounting properly for the time it takes to perform the callback, which I've seen get missed a lot.

I'll leave it up to your judgement as to whether we want to classify this as a bugfix or an improvement in the changelog. If it's a bugfix, we should backport it.

File: helper/exptime/backoff.go
Comment (shoenig) [reply helper/exptime/backoff.go:63]: :+1: added the license

But yeah, the only reason I don't want to pull it in as a module is the transitive dependency on github.com/gojuno/minimock/v3 is several orders of magnitude larger than this one function we actually want

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**
